yb-voyager/src/callhome/diagnostics.go

```diff
@@ -349,8 +353,44 @@ func SendPayload(payload *Payload) error {
 
 // We want to ensure that no user-specific information is sent to the call-home service.
 // Therefore, we only send the segment of the error message before the first ":" as that is the generic error message.
-// Note: This is a temporary solution. A better solution would be to have
-// properly structured errors and only send the generic error message to callhome.
-func SanitizeErrorMsg(errorMsg string) string {
-	return strings.Split(errorMsg, ":")[0]
+// Accepts error type, returns empty string if error is nil.
+func SanitizeErrorMsg(err error) string {
+	if err == nil {
+		return ""
+	}
+	errorMsg := strings.Split(err.Error(), ":")[0]
+	additionalContext := getSpecificContextForError(err)
+	if additionalContext != nil {
+		var contextData string
+		contextJsonBytes, err := json.Marshal(additionalContext)
+		if err != nil {
+			// should not happen.
+			log.Errorf("callhome: error while marshalling additional context for error: %v", err)
+		}
+		contextData = string(contextJsonBytes)
+		errorMsg = fmt.Sprintf("%s: %s", errorMsg, contextData)
+	}
+	return errorMsg
+}
+
+func getSpecificContextForError(err error) map[string]string {
+	if err == nil {
+		return nil
+	}
+	context := make(map[string]string)
+
+	var ibe errs.ImportBatchError
+	if errors.As(err, &ibe) {
+		context["step"] = ibe.Step()
+		context["flow"] = ibe.Flow()
+	}
+
+	var pgErr *pgconn.PgError
+	if errors.As(err, &pgErr) {
+		// If the error is a pgconn.PgError, we can return a more
+		// specific error message that includes the SQLSTATE code
+		context["pg_error_code"] = pgErr.Code
```

Comment:
Can we use the `dbSpecficContext` map to store this, and check it in the map directly?

Reply:
I thought about that, but checking for pgError here will also help in other errors (not just ImportBatchErrors). 

the dbspecificContext is useful in cases where the default error String representation lacks certain details. 
For example, for the COPY error, the **default** string representation is just `ERROR: duplicate key value violates unique constraint "test_uk_email_key" (SQLSTATE 23505)`. (notice that it already has the code). We specifically fetch `COPY test_uk, line 11: ""` from the `pgErr` struct so as to add more context to the `ImportBatchError`.